src/caja-window.c

```diff
@@ -2170,7 +2185,18 @@ caja_window_class_init (CajaWindowClass *class)
     GtkBindingSet *binding_set;
 
     G_OBJECT_CLASS (class)->constructor = caja_window_constructor;
-    G_OBJECT_CLASS (class)->constructed = caja_window_constructed;
+#ifdef HAVE_WAYLAND
+    /*For some reason if we use the constructed() class method
+     *in wayland with the compositor set to or forcing CSD
+     *we lose the ability to drag the window by the titlebar
+     */
```

Comment:
I would guess then that some of the things done in `constructed()` are done too late for some compositors.  If that's the case, can't we simply do it earlier for everyone, instead of having that weird setup?

Reply:
That is exactly what the latest force-push does: removes the constructed() class method entirely and moves setting initial geometry and bookmarks to the constructor() class